src/server/Demo.Web/Controllers/TestController.cs

```diff
@@ -16,11 +16,11 @@ public TestController(IRepository<Foo> fooRepository)
         }
 
         [HttpGet]
-        public async Task<ActionResult<IEnumerable<Foo>>> Test()
+        public ActionResult<IAsyncEnumerable<Foo>> Test()
         {
             try
             {
-                var result = await _fooRepository.GetAll();
+                var result = _fooRepository.GetAll();
                 return Ok(result);
             }
             catch (Exception ex)
```

Comment:
Not related to this particular PR, but caught my eye - an explicit exception handling only makes sense if we have any additional info to log / add to exception details. But that's not the case here, so we should let the method `throw` and let the general pipeline exception handler handle it (respond with 500). BTW, `Problem(ex.Message, null, 500)` seems to be the same as `Problem(ex.Message)`.

Reply:
Indeed it is in this. It was meant to be a basis for testing and returning errors for the FE to test against (could be extended with a mechanism to return different error codes etc, but I did not go that far). Can be removed if not needed.